### ISSUE TITLE
Monitor memory and CPU usage in AIT 

### ISSUE THREAD

**Issue by Selutario** [wazuh/wazuh#23110]
## Description

One of the hypotheses behind the problem in `modulesd` seen in the issue below is that the process dies due to lack of memory (OOM):
- https://github.com/wazuh/wazuh/issues/21748

We decided to manually run the tests in a t3.medium host while monitoring the resources usage, which should be done as part of this issue. 

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [x] Tests (unit tests, API integration tests).
- [x] Changelog.
- [x] Documentation.
- [x] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by GGP1** [comment]
## Update

<details><summary>Instance details</summary>

![instance](https://github.com/wazuh/wazuh/assets/51374959/302d433b-638f-498d-a48d-1810d2f60362)

</details>

<details><summary>OS details</summary>

```console
root@ip-172-31-41-3:/home/ubuntu# cat /etc/os-release
PRETTY_NAME="Ubuntu 24.04 LTS"
NAME="Ubuntu"
VERSION_ID="24.04"
VERSION="24.04 LTS (Noble Numbat)"
VERSION_CODENAME=noble
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=noble
LOGO=ubuntu-logo
```

</details>

<details><summary>CPU details</summary>

```console
root@ip-172-31-41-3:/home/ubuntu# cat /proc/cpuinfo
processor	: 0
vendor_id	: AuthenticAMD
cpu family	: 23
model		: 1
model name	: AMD EPYC 7571
stepping	: 2
microcode	: 0x800126e
cpu MHz		: 2199.996
cache size	: 512 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 1
apicid		: 0
initial apicid	: 0
fpu		: yes
fpu_exception	: yes
cpuid level	: 13
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch topoext vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 clzero xsaveerptr arat npt nrip_save
bugs		: sysret_ss_attrs null_seg spectre_v1 spectre_v2 spec_store_bypass retbleed smt_rsb srso div0
bogomips	: 4399.99
TLB size	: 2560 4K pages
clflush size	: 64
cache_alignment	: 64
address sizes	: 48 bits physical, 48 bits virtual
power management:

processor	: 1
vendor_id	: AuthenticAMD
cpu family	: 23
model		: 1
model name	: AMD EPYC 7571
stepping	: 2
microcode	: 0x800126e
cpu MHz		: 2199.996
cache size	: 512 KB
physical id	: 0
siblings	: 2
core id		: 0
cpu cores	: 1
apicid		: 1
initial apicid	: 1
fpu		: yes
fpu_exception	: yes
cpuid level	: 13
wp		: yes
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 sse4_1 sse4_2 movbe popcnt aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy cr8_legacy abm sse4a misalignsse 3dnowprefetch topoext vmmcall fsgsbase bmi1 avx2 smep bmi2 rdseed adx smap clflushopt sha_ni xsaveopt xsavec xgetbv1 clzero xsaveerptr arat npt nrip_save
bugs		: sysret_ss_attrs null_seg spectre_v1 spectre_v2 spec_store_bypass retbleed smt_rsb srso div0
bogomips	: 4399.99
TLB size	: 2560 4K pages
clflush size	: 64
cache_alignment	: 64
address sizes	: 48 bits physical, 48 bits virtual
power management:
```

</details>

<details><summary>RAM details</summary>

```console
root@ip-172-31-41-3:/home/ubuntu# free -h
               total        used        free      shared  buff/cache   available
Mem:           3.8Gi       456Mi       2.7Gi       2.7Mi       850Mi       3.3Gi
Swap:             0B          0B          0B
root@ip-172-31-41-3:/home/ubuntu# dmidecode -t memory
# dmidecode 3.5
Getting SMBIOS data from sysfs.
SMBIOS 2.7 present.

Handle 0x0008, DMI type 16, 23 bytes
Physical Memory Array
	Location: System Board Or Motherboard
	Use: System Memory
	Error Correction Type: Unknown
	Maximum Capacity: 4 GB
	Error Information Handle: Not Provided
	Number Of Devices: 1

Handle 0x0009, DMI type 17, 34 bytes
Memory Device
	Array Handle: 0x0008
	Error Information Handle: Not Provided
	Total Width: 128 bits
	Data Width: 64 bits
	Size: 4 GB
	Form Factor: DIMM
	Set: None
	Locator: Not Specified
	Bank Locator: Not Specified
	Type: DDR4
	Type Detail: Static Column Pseudo-static Synchronous Window DRAM
	Speed: 2667 MT/s
	Manufacturer: Not Specified
	Serial Number: Not Specified
	Asset Tag: Not Specified
	Part Number: Not Specified
	Rank: Unknown
	Configured Memory Speed: Unknown
root@ip-172-31-41-3:/home/ubuntu# lshw -short -C memory
H/W path       Device           Class          Description
==========================================================
/0/0                            memory         64KiB BIOS
/0/4/5                          memory         3MiB L1 cache
/0/4/6                          memory         16MiB L2 cache
/0/4/7                          memory         64MiB L3 cache
/0/8                            memory         4GiB System Memory
/0/8/0                          memory         4GiB DIMM DDR4 Static column Pseudo-static Synchronous Window DRAM 2667 MHz (0.4 ns)
```

</details>

<details><summary>Python version and dependencies</summary>

```console
root@ip-172-31-41-3:/home/ubuntu/Python-3.10.0# python3.10 --version
Python 3.10.12
root@ip-172-31-41-3:/home/ubuntu/wazuh/framework# python3.10 -m pip --version
pip 23.0.1 from /usr/local/lib/python3.10/site-packages/pip (python 3.10)
root@ip-172-31-41-3:/home/ubuntu/wazuh/framework# python3.10 -m pip freeze
aiohttp==3.9.3
aiohttp-cache==2.2.0
aiohttp-cors==0.7.0
aiohttp-jinja2==1.5.1
aioredis==1.3.1
aiosignal==1.2.0
asn1crypto==1.3.0
async-generator==1.10
async-timeout==4.0.2
attrs==20.3.0
azure-common==1.1.25
azure-storage-blob==2.1.0
azure-storage-common==2.1.0
boto3==1.17.85
botocore==1.20.85
cachetools==4.1.0
certifi==2023.7.22
cffi==1.15.1
chardet==3.0.4
charset-normalizer==2.0.4
click==8.1.3
clickclick==20.10.2
connexion==2.14.2
cryptography==42.0.4
Cython==0.29.21
defusedxml==0.6.0
docker==6.0.0
docker-pycreds==0.4.0
docopt==0.6.2
docutils==0.15.2
envparse==0.2.0
exceptiongroup==1.2.1
Flask==2.2.5
freezegun==0.3.15
frozenlist==1.2.0
future==0.18.3
google-api-core==1.30.0
google-auth==1.28.0
google-cloud-core==1.7.1
google-cloud-pubsub==2.7.1
google-cloud-storage==1.39.0
google-crc32c==1.1.2
google-resumable-media==1.3.1
googleapis-common-protos==1.51.0
greenlet==2.0.2
grpc-google-iam-v1==0.12.3
grpcio==1.58.0
hiredis==2.2.3
idna==2.9
importlib-metadata==3.10.1
inflection==0.3.1
iniconfig==2.0.0
itsdangerous==2.0.0
Jinja2==3.1.3
jmespath==0.9.5
jsonschema==2.6.0
libcst==0.3.20
MarkupSafe==2.1.2
more-itertools==8.2.0
multidict==5.2.0
mypy-extensions==0.4.3
numpy==1.26.0
openapi-spec-validator==0.2.6
outcome==1.3.0.post0
packaging==20.9
paho-mqtt==1.5.1
pathlib==1.0.1
pbr==6.0.0
pluggy==1.5.0
proto-plus==1.19.0
protobuf==3.19.6
psutil==5.9.0
py==1.11.0
pyarrow==14.0.1
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycparser==2.21
PyJWT==2.8.0
pykwalify==1.8.0
pyparsing==2.4.7
pytest==7.3.1
pytest-aiohttp==1.0.4
pytest-asyncio==0.18.1
pytest-html==2.1.1
pytest-metadata==3.1.1
pytest-trio==0.8.0
python-box==5.4.1
python-dateutil==2.8.1
python-json-logger==2.0.2
pytz==2020.1
PyYAML==5.3.1
requests==2.31.0
rsa==4.7.2
ruamel.yaml==0.18.6
ruamel.yaml.clib==0.2.8
s3transfer==0.4.2
secure==0.2.1
six==1.16.0
sniffio==1.3.1
sortedcontainers==2.4.0
SQLAlchemy==2.0.23
stevedore==3.5.2
tabulate==0.8.9
tavern==1.23.5
tomli==2.0.1
trio==0.24.0
typing-inspect==0.7.1
typing_extensions==4.5.0
urllib3==1.26.18
uvloop==0.17.0
websocket-client==0.57.0
Werkzeug==2.2.3
xmltodict==0.12.0
yarl==1.7.0
zipp==3.3.2
```

</details>

<details><summary>Docker and docker compose versions</summary>

```console
root@ip-172-31-41-3:/home/ubuntu/wazuh/api/test/integration# docker --version
Docker version 26.1.0, build 9714adc
root@ip-172-31-41-3:/home/ubuntu/wazuh/api/test/integration# docker compose version
Docker Compose version v2.26.1
```

</details>

<details><summary>Docker image building</summary>

```console
root@ip-172-31-41-3:/home/ubuntu/wazuh/api/test/integration/env# docker compose --profile cluster build --build-arg WAZUH_BRANCH=4.8.0
...
root@ip-172-31-41-3:/home/ubuntu/wazuh/api/test/integration/env# docker images
REPOSITORY                         TAG       IMAGE ID       CREATED              SIZE
<none>                             <none>    5b716e877e8d   5 seconds ago        160MB
integration_test_wazuh-agent_old   latest    edcf66f9a468   5 seconds ago        160MB
<none>                             <none>    3b3df51356e4   5 seconds ago        160MB
<none>                             <none>    60bf64be03cd   5 seconds ago        160MB
<none>                             <none>    a7aafa553797   44 seconds ago       1.2GB
<none>                             <none>    3f64fecb6391   44 seconds ago       1.2GB
integration_test_wazuh-agent       latest    9399ccbf255e   44 seconds ago       1.2GB
integration_test_wazuh-manager     latest    ac878ac5efb7   About a minute ago   2.28GB
<none>                             <none>    f9f628e61240   About a minute ago   2.28GB
<none>                             <none>    657c48c3ed3b   About a minute ago   2.28GB
<none>                             <none>    333bc135e203   12 minutes ago       1.2GB
integration_test_nginx-lb          latest    cab05137413a   23 minutes ago       188MB
```

</details>

<details><summary>Tests execution</summary>

```console
root@ip-172-31-41-3:/home/ubuntu/wazuh/api/test/integration# python3.10 -m pytest --nobuild test_*.tavern.yaml
================================================================= test session starts ==================================================================
platform linux -- Python 3.10.12, pytest-7.3.1, pluggy-1.5.0
rootdir: /home/ubuntu/wazuh/api/test/integration
configfile: pytest.ini
plugins: html-2.1.1, metadata-3.1.1, trio-0.8.0, asyncio-0.18.1, aiohttp-1.0.4, tavern-1.23.5
asyncio: mode=auto
collected 1063 items                                                                                                                                   

test_active_response_endpoints.tavern.yaml ..                                                                                                    [  0%]
test_agent_DELETE_endpoints.tavern.yaml ......                                                                                                   [  0%]
...
```

</details>

<details><summary>Statistics</summary>

### Window 1

```console
root@ip-172-31-41-3:/home/ubuntu# sar -r -u 30 -o /data > /dev/null 2>&1
```

### Window 2

#### CPU

```console
root@ip-172-31-41-3:/home/ubuntu# sar -u -f /data
Linux 6.8.0-1008-aws (ip-172-31-41-3) 	04/26/24 	_x86_64_	(2 CPU)

20:23:26        CPU     %user     %nice   %system   %iowait    %steal     %idle
20:23:56        all     53.51     16.10     29.29      0.00      1.10      0.00
20:24:26        all     58.71      9.38     31.86      0.00      0.05      0.00
20:24:56        all     51.37     12.82     35.81      0.00      0.00      0.00
20:25:26        all     36.94     19.05     43.90      0.00      0.12      0.00
20:25:56        all     37.30     14.61     48.01      0.00      0.08      0.00
20:26:26        all     30.56     15.49     53.96      0.00      0.00      0.00
20:26:56        all     37.27     24.56     38.17      0.00      0.00      0.00
20:27:26        all     44.24      8.43     47.31      0.00      0.02      0.00
20:27:56        all     24.25     16.14     59.61      0.00      0.00      0.00
20:28:26        all     30.74      4.79     40.97      0.05      5.42     18.03
20:28:56        all     17.35     24.13     55.04      0.00      3.48      0.00
20:29:26        all     21.16     19.71     59.14      0.00      0.00      0.00
20:29:56        all     24.66     10.98     61.81      0.00      0.65      1.90
20:30:26        all     26.85      8.67     49.54      0.02      6.29      8.64
20:30:56        all     24.24      3.75     47.06      0.15      6.72     18.07
20:31:26        all     26.25      8.22     34.58      0.03      6.77     24.15
20:31:56        all     23.94      0.32     30.35      0.03      9.61     35.75
20:32:26        all     48.02      0.00     24.82      0.00      5.02     22.14
20:32:56        all     58.64      0.02     26.15      0.00      6.07      9.12
20:33:26        all     57.53      0.00     25.72      0.00      4.95     11.79
20:33:56        all     60.15      0.02     26.65      0.00      3.34      9.84
20:34:26        all     43.18      0.03     26.38      0.00      7.51     22.89
20:34:56        all     24.90      0.02     24.06      0.00      7.31     43.72
20:35:26        all     23.38      0.03     24.23      0.05      6.42     45.89
20:35:56        all     23.17      0.00     23.19      0.03      9.01     44.59
20:36:26        all     26.20      0.03     24.68      0.02      6.13     42.94
20:36:56        all     23.09      0.03     23.98      0.00      9.50     43.39
20:37:26        all     24.95      0.00     26.71      0.05      8.02     40.26
20:37:56        all     27.22      0.00     25.20      0.00      9.01     38.58
20:38:26        all     26.55      0.00     28.99      0.00      6.66     37.80
20:38:56        all     26.01      0.03     27.14      0.03      7.74     39.04
20:39:26        all     26.57      0.00     27.91      0.00      8.31     37.21
20:39:56        all     27.67      0.00     27.41      0.02     15.63     29.27
20:40:26        all     26.86      0.02     27.83      0.00      9.60     35.69
20:40:56        all     25.31      0.02     26.13      0.02      6.71     41.82
20:41:26        all     29.32      0.00     26.99      0.02      7.18     36.49
20:41:56        all     26.60      0.03     27.41      0.00      6.38     39.58
20:42:26        all     26.89      0.02     27.51      0.02      8.15     37.42
20:42:56        all     26.28      0.00     27.23      0.02      8.80     37.67
20:43:26        all     28.03      0.02     28.08      0.00      9.51     34.37
20:43:56        all     25.93      0.00     26.85      0.02      9.45     37.77
20:44:26        all     26.22      0.00     27.13      0.03     11.65     34.97
20:44:56        all     27.94      0.03     26.89      0.00     10.82     34.31
20:45:26        all     26.06      0.00     27.28      0.02     10.71     35.93
20:45:56        all     26.93      0.00     29.01      0.02      8.77     35.28
20:46:26        all     27.48      0.03     27.48      0.02      8.43     36.56
20:46:56        all     39.96      0.02     26.65      0.00     10.14     23.24
20:47:26        all     59.37      0.03     25.58      0.00      6.64      8.38
20:47:56        all     34.29      0.00     25.23      0.03      6.64     33.82
20:48:26        all     40.14      0.00     25.12      0.02      6.00     28.72
20:48:56        all     58.40      0.00     26.85      0.02      6.86      7.88
20:49:26        all     57.24      0.00     25.41      0.00      7.03     10.31
20:49:56        all     60.74      0.03     24.84      0.07      5.39      8.93
20:50:26        all     67.04      6.37     23.32      0.00      3.27      0.00
20:50:56        all     30.82     11.94     47.62      0.00      1.38      8.24
20:51:26        all     22.78      0.02     23.73      0.05      8.80     44.63
20:51:56        all     22.60      0.02     24.19      0.02      7.65     45.53
20:52:26        all     49.02      0.02     25.06      0.00      6.51     19.39
20:52:56        all     76.43      3.56     17.09      0.02      0.43      2.47
20:53:26        all     51.74     17.02     31.24      0.00      0.00      0.00
20:53:56        all     22.11      7.67     52.82      0.00      1.52     15.88
20:54:26        all     23.59      0.00     24.73      0.00      8.44     43.24
20:54:56        all     42.35      0.00     25.57      0.03      7.89     24.15
20:55:26        all     59.12      0.02     27.40      0.00      5.76      7.70
20:55:56        all     59.66      0.00     26.33      0.00      5.73      8.29
20:56:26        all     59.15      0.02     25.09      0.00      7.01      8.74
20:56:56        all     58.80      0.00     26.63      0.00      5.11      9.46
20:57:26        all     61.37      0.02     26.89      0.00      3.76      7.96

Average:        all     37.11      3.89     31.52      0.01      5.87     21.59
```

#### Memory

```console
root@ip-172-31-41-3:/home/ubuntu# sar -r -f /data
Linux 6.8.0-1008-aws (ip-172-31-41-3) 	04/26/24 	_x86_64_	(2 CPU)

20:23:26    kbmemfree   kbavail kbmemused  %memused kbbuffers  kbcached  kbcommit   %commit  kbactive   kbinact   kbdirty
20:23:56      1304708   2696724    850120     21.57    109572   1270676   4446632    112.82   1283960    796612      3436
20:24:26      1064364   2490988   1049324     26.62    114708   1293908   6731928    170.80   1488020    824644      8508
20:24:56       829076   2333456   1204072     30.55    116236   1351352   7611176    193.11   1647904    883524      8912
20:25:26       637032   2183608   1350828     34.27    116856   1380476   8723212    221.33   1787348    913012     17680
20:25:56       491724   2105624   1427168     36.21    117588   1432308   8653724    219.56   1875984    954104     16792
20:26:26       414272   2041860   1490572     37.82    117776   1445616   8978172    227.80   1952916    967592      4028
20:26:56       329180   1970400   1561732     39.62    117932   1455016   8907552    226.00   1985952    977144      6864
20:27:26       261016   1937032   1592752     40.41    118056   1465128   8576680    217.61   2069956    987388      3680
20:27:56       212376   1904468   1625088     41.23    118220   1480304   8950204    227.09   2050932   1002596     13360
20:28:26       216224   1908880   1620304     41.11    118764   1481100   8732676    221.57   2085044   1003944      4192
20:28:56       149068   1859552   1669436     42.36    118912   1498320   9664812    245.22   2119544   1021316      5756
20:29:26       232232   1861244   1667572     42.31    106900   1429020   9226952    234.11   2220056    846628      5784
20:29:56       280704   1918932   1609324     40.83    107192   1436804   9006200    228.51   2173332    856380     18524
20:30:26       231652   1876408   1652488     41.93    107412   1443928   9152328    232.22   2193332    860588      9832
20:30:56       208904   1854412   1674344     42.48    107576   1444548   9233276    234.27   2217008    861380      6404
20:31:26       221652   1866876   1661924     42.17    107736   1444220   9247672    234.63   2211760    861220      5488
20:31:56       231516   1874016   1654764     41.99    107768   1441428   9040684    229.38   2205844    858428       684
20:32:26       205032   1844212   1684112     42.73    107808   1438060   9060668    229.89   2223548    854964       552
20:32:56       174728   1814016   1713196     43.47    107832   1438116   9306212    236.12   2266404    855056       864
20:33:26       155344   1794892   1731700     43.94    107904   1438248   9116460    231.30   2267824    855076       812
20:33:56       140112   1780000   1745960     44.30    107924   1438500   9233216    234.27   2280144    855160      1204
20:34:26       155660   1795756   1730164     43.90    107944   1438636   9136164    231.80   2284380    855312       816
20:34:56       144452   1784576   1741312     44.18    107960   1438612   9149516    232.14   2296024    855304       952
20:35:26       140584   1780704   1745140     44.28    107984   1438536   9160740    232.43   2302116    855252      1200
20:35:56       136736   1777012   1748920     44.37    107996   1438640   9135808    231.80   2281720    855244       476
20:36:26       140992   1781460   1744540     44.26    108024   1438756   9137152    231.83   2283444    855388      1120
20:36:56       129724   1770372   1755584     44.54    108040   1438880   9260724    234.97   2296796    855520       596
20:37:26       135516   1776240   1749748     44.40    108092   1438860   9136332    231.81   2283552    855560       908
20:37:56       126596   1767480   1758496     44.62    108132   1438948   9150372    232.17   2295496    855684      1016
20:38:26       237436   1776808   1749212     44.38    102704   1343452   9133632    231.74   2323056    713204       820
20:38:56       231160   1770616   1755364     44.54    102712   1343512   9409856    238.75   2331220    713272       804
20:39:26       229108   1768784   1757196     44.58    102716   1343724   9149040    232.13   2335188    713476       680
20:39:56       228904   1768864   1757068     44.58    102724   1343984   9138552    231.87   2327388    713536       948
20:40:26       220368   1760432   1765452     44.79    102728   1344080   9144256    232.01   2332696    713628       528
20:40:56       216924   1757188   1768804     44.88    102736   1344264   9315580    236.36   2333940    713696       960
20:41:26       236328   1776692   1749236     44.38    102740   1344340   9142076    231.95   2327372    713776       572
20:41:56       219932   1760556   1765380     44.79    102748   1344548   9344452    237.09   2351352    713856       924
20:42:26       214840   1755572   1770308     44.92    102772   1344588   9461252    240.05   2366216    713920       768
20:42:56       216128   1756956   1768976     44.88    102788   1344640   9151676    232.20   2337128    713984       516
20:43:26       226376   1767520   1758128     44.61    102816   1344836   9158512    232.37   2339196    714148       896
20:43:56       221060   1762572   1763032     44.73    102816   1345160   9163516    232.50   2340792    714340       692
20:44:26       207796   1749412   1776188     45.07    102820   1345220   9168312    232.62   2344660    714404       504
20:44:56       200172   1742068   1783392     45.25    102824   1345448   9182808    232.99   2352472    714452       660
20:45:26       222548   1764568   1760960     44.68    102824   1345544   9363728    237.58   2352328    714540       328
20:45:56       207868   1749988   1775508     45.05    102844   1345572   9450012    239.77   2358500    714600       676
20:46:26       230944   1773496   1752032     44.45    103016   1345784   9179328    232.90   2347356    714856       304
20:46:56       204608   1747460   1778192     45.12    103156   1345896   9189408    233.16   2356156    715100       320
20:47:26       150288   1693408   1831888     46.48    103332   1346064   9459460    240.01   2427856    715340       952
20:47:56       196812   1740304   1785224     45.30    103580   1346004   9227044    234.11   2372648    715632       684
20:48:26       197560   1741272   1784236     45.27    103692   1346068   9214732    233.80   2363288    715816       744
20:48:56       191528   1735876   1789656     45.41    103848   1346460   9224000    234.03   2371568    716232      1196
20:49:26       141376   1685952   1839496     46.67    103920   1346580   9235832    234.33   2435932    716420       796
20:49:56       814644   2254844   1272632     32.29    102804   1245888   5986216    151.88   1677836    780924      1172
20:50:26       384780   1825748   1699756     43.13    102884   1246952   8824536    223.90   2111196    781512      1768
20:50:56       389220   1835960   1689404     42.86    102888   1252576   8944008    226.93   2114388    787156      2072
20:51:26       369380   1816064   1709344     43.37    102912   1252500   8962128    227.39   2128516    786976       460
20:51:56       364348   1811620   1713692     43.48    103012   1252708   8952512    227.15   2123828    787164      1004
20:52:26       720768   2167936   1359168     34.49    103144   1252360   5756128    146.05   1779188    787236       368
20:52:56       562512   2008616   1517300     38.50    103308   1251704   8568796    217.41   1916140    786500      1828
20:53:26       438336   1886748   1638724     41.58    103320   1254056   8971740    227.63   2064588    788812      2736
20:53:56       484804   1937592   1587908     40.29    103380   1258292   8884452    225.42   1994052    792988       996
20:54:26       477148   1930000   1595340     40.48    103384   1258340   8896228    225.72   2003172    792888       768
20:54:56       404248   1857292   1668012     42.32    103448   1258428   9136288    231.81   2097988    793036       856
20:55:26       383668   1836840   1688292     42.84    103456   1258504   9321508    236.51   2104384    793120       876
20:55:56       384660   1837944   1687168     42.81    103460   1258560   9326732    236.64   2111388    793180       400
20:56:26       390900   1844432   1680720     42.64    103464   1258764   9130328    231.66   2100636    793264       716
20:56:56       371772   1825416   1699668     43.12    103468   1258836   9151160    232.19   2110820    793328       744

Average:       311827   1867681   1659742     42.11    106597   1361361   8899807    225.81   2171594    806124      2738
```

</details>

### Notes

- Had to upgrade `pytest-trio` to `0.8.0` for the tests to work, otherwise I get the error `AttributeError: module 'trio' has no attribute 'MultiError'`

- Had to downgrade `PyYAML` to `5.3.1` to avoid the error `AttributeError: cython_sources` when installing the dependencies.

- CPU usage increases on heavier tests, such as `test_cluster_endpoints`, specially when the test restarts all services. On the other hand, RAM usage remains the same.

### Conclusion

I didn't notice a high resource consumption during the tests execution, the average CPU usage was `37.11` and the average memory usage `42.11`. 

Furthermore, the stats were gathered using a single instance whereas the pipeline uses one instance per test, reducing the overall load.

However, it is worth mentioning that the memory `%commit` values were higher than 100%, meaning that if all processes requesting memory all touch the memory the kernel says they can, the system would run out of memory. But these values are not uncommon, sometimes it ranges between even higher values. See [Understanding ‘commit%’ in Sar](http://www.throwingbones.com/ben/blog/2020-06-15-commit-pct-in-sar/index.html) for more information.

>  `%commit` is the percentage of RAM the kernel has committed to processes.

In summary, I don't think the instances used for testing are resource-constrained, they have enough RAM and CPU to setup the environment and run any of our tests. 

There could be sporadic and rare cases in which multiple processes claim the RAM promised by the kernel and that causes an out of memory error. but it shouldn't be something that happens as regularly as the `wazuh-modulesd` error.

**Comment by fdalmaup** [comment]
# Review

The analysis regarding resource use was thorough and detailed. When the manager processes don't show abnormal behavior, CPU, and memory usage for the AITs execution remains normal. No modification is needed regarding the environment created to run the tests.
The only question that arises and that is not part of the issue's objective is if a mechanism to track the origin of sporadic error/misbehavior cases should be implemented so that solution proposals can be made depending on the affected component, avoiding the re-launch of the tests. 

LGTM!

**Comment by Selutario** [comment]
Nice job. We will need to reopen https://github.com/wazuh/wazuh/issues/21748 as the problem still exists.
